File: Solutions/0169/0169.ts
function majorityElement(nums: number[]): number {
    let count = 0;
    let ans: number;

    for (const num of nums) {
        if (count === 0) {
            ans = num;
        }
        count += ans === num ? 1 : -1;
    }

    return ans;
};

File: Solutions/0169/0169.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
       int count=0;
       int ans;
       for(const int num:nums){
           if(count==0)
               ans=num;
            count+=ans==num?1:-1;
       }
       return ans;

    }
};